src/platform/plugins/private/presentation_panel/public/panel_component/presentation_panel_error_internal.tsx

```diff
@@ -80,20 +80,26 @@ export const PresentationPanelErrorInternal = ({ api, error }: PresentationPanel
     const subscription = new Subscription();
     (async () => {
       const editPanelAction = await uiActions.getAction(ACTION_EDIT_PANEL);
-      if (canceled || !editPanelAction?.couldBecomeCompatible?.({ embeddable: api })) return;
-
-      const initiallyCompatible = await editPanelAction?.isCompatible({
+      const context = {
         embeddable: api,
         trigger: { id: CONTEXT_MENU_TRIGGER },
-      } as EmbeddableApiContext & ActionExecutionMeta);
+      };
+      if (canceled || !editPanelAction?.couldBecomeCompatible?.(context)) return;
+
+      const initiallyCompatible = await editPanelAction?.isCompatible(context);
```

Comment:
Is there any chance that the error API starts out with no edit method, then eventually gets one? It seems to me that this maybe shouldn't have used the `subscribeToCompatibilityChanges` method to begin with.

Reply:
> Is there any chance that the error API starts out with no edit method, then eventually gets one?

Yes. The edit action compatibility can change if the dashboard is opened in view mode and then switched to edit mode.

Reply:
Ah yes, makes sense!